From 40c2b1606c44cb8e836ae4fa78ae6a415e153e28 Mon Sep 17 00:00:00 2001
From: Varun Sharma <vsharma@habana.ai>
Date: Tue, 29 Nov 2022 10:24:03 +0200
Subject: [PATCH] [SW-0] greco: read security enable from efuse

Instead of reading security enable config from sp sram,
read it directly from efuse security offset.

Change-Id: I04e8fc905543b9cfb1c1cd4dbb189e7588738c34
Signed-off-by: Varun Sharma <vsharma@habana.ai>
---

diff --git a/arch/greco/arc/security.c b/arch/greco/arc/security.c
index 08c4ef7..df67a07 100644
--- a/arch/greco/arc/security.c
+++ b/arch/greco/arc/security.c
@@ -12,6 +12,7 @@
 #include <sha256.h>
 #include <greco/greco_security.h>
 #include <hl.h>
+#include <efuse.h>
 
 #define EFUSE_SYSCFG_ENABLE_VALIDATION_MASK	BIT(10)
 
@@ -24,8 +25,12 @@
 bool preboot_security_is_enabled(void)
 {
 	u32 config_efuse;
+	int ret;
 
-	config_efuse = sp_sram_readl(SECURITY_EFUSE_SYS_CFG_OFFSET);
+	ret = preboot_fuse_read(EFUSE_SYSCFG_OFFSET, &config_efuse);
+	if (ret)
+		return false;
+
 	return (config_efuse & EFUSE_SYSCFG_ENABLE_VALIDATION_MASK);
 }
 
